internal/clients/manager_test.go

```diff
@@ -428,21 +460,89 @@ func TestAdminAPIClientsManager_ConcurrentNotify(t *testing.T) {
 	}()
 
 	// Run multiple notifiers in parallel to make sure that Notify is safe for concurrent use.
+	clients := []adminapi.DiscoveredAdminAPI{
+		testDiscoveredAdminAPI("http://10.0.0.1:8080"),
+	}
 	for i := 0; i < 10; i++ {
+		i := i
 		go func() {
-			m.Notify([]adminapi.DiscoveredAdminAPI{
-				testDiscoveredAdminAPI("http://10.0.0.1:8080"),
-			})
+			// Notify with different clients' sets to trigger notifications as changes are required for them to happen.
+			if i%2 == 0 {
```

Comment:
The mod 2 thing seems a bit magic. Can we add more comments or rewrite this a bit so that it's easier to comprehend?

Reply:
Yeah, I rewrote it so maybe now it's more direct - we have two sets of clients: odd and even, and we use them interchangeably to cause an actual change that will trigger a notification. PTAL